viz/RobotModel.cpp

```diff
@@ -399,21 +619,172 @@ osg::ref_ptr<osg::Node> RobotModel::makeOsg( boost::shared_ptr<urdf::ModelInterf
+            hook = hook_buffer.back();
+            hook_buffer.pop_back();
+
+            osg::Node* new_hook = makeOsg2(kdl_segment, sdf_link, sdf_parent_link, hook->asGroup());
+
+            //Append hooks
+            for(uint i = 0; i < it->second.children.size(); i++){
+                hook_buffer.push_back(new_hook);
+            }
+
+            std::cout << "segment_name: " << kdl_segment.getName() << std::endl;
+
+            segmentNames_.push_back(kdl_segment.getName());
+            if(kdl_segment.getJoint().getType() != KDL::Joint::None)
+                jointNames_.push_back(kdl_segment.getJoint().getName());
+        }
+
+    }
+
+    /**
+     * in the SDF files it is not necessary one joint per two links
+     * in the SDF files it is possible to create a file with multiple links without joints
+     * original_root_ is not a segment, is only used to store the segments
+     * if the links in the SDF doesn't have a joint, then this segment will be a direct child of original_root_
+     * if all links doen't have a joint, then all segments will be children direct of original_root_
+     */
+    root_->addChild(original_root_);
+    return root_;
+}
+
 osg::ref_ptr<osg::Node> RobotModel::load(QString path){
 
+    loadPlugins();
     loadEmptyScene();
 
+    QString suffix = QFileInfo(path).suffix().toLower();
+
+    /*
+     * call the function based in the file suffix
+     */
+    if (loadFunctions.contains(suffix)){
+        osg::Node *node= CALL_MEMBER_POINTER_FUNCTION(*this, loadFunctions[suffix])(path);
+        return node;
+    }
+    else {
+        LOG_ERROR("the %s type of file is not supported .", suffix.toStdString().c_str());
+    }
+
+    return new osg::Group();
+}
+
+osg::Node* RobotModel::loadURDF(QString path)
+{
     std::ifstream t( path.toStdString().c_str() );
     std::string xml_str((std::istreambuf_iterator<char>(t)),
-	                     std::istreambuf_iterator<char>());
+                       std::istreambuf_iterator<char>());
+
+    rootPrefix = QDir(QFileInfo(path).absoluteDir());
+
     //Parse urdf
     boost::shared_ptr<urdf::ModelInterface> model = urdf::parseURDF( xml_str );
-    rootPrefix = QDir(QFileInfo(path).absoluteDir());
     if (!model)
         return NULL;
+
     return makeOsg(model);
 }
 
+osg::Node* RobotModel::loadSDF(QString path)
+{
+    rootPrefix = QDir(QFileInfo(path).absoluteDir());
+
+    sdf::SDFPtr sdf(new sdf::SDF);
+
+    if (!sdf::init(sdf)){
+        LOG_ERROR("unable to initialize sdf.");
+        return NULL;
+    }
+
+    if (!sdf::readFile(path.toStdString(), sdf)){
+        LOG_ERROR("unabled to read sdf file %s.", path.toStdString().c_str());
+        return NULL;
+    }
+
+    if (!sdf->root->HasElement("model")){
+        LOG_ERROR("the <model> tag not exists");
+        return NULL;
+    }
+
+    return makeOsg(sdf->root->GetElement("model"));
+}
+
+void RobotModel::loadPlugins()
+{
+    osgDB::FileNameList plugins = osgDB::listAllAvailablePlugins();
```

Comment:
It is necessary to call loadPlugin function to load COLLADA meshes.
If we don't call it, we get this error: "Could not find plugin to read objects from file".